### PR TITLE
Fixed #37, improved PERSON entity splitting (consistent with X-Ent)

### LINKED ISSUE #37: Fix splitting of PERSON named entities that contain `-`
Look in annotations for examples

### PR DESCRIPTION
**Pull request by anton164**
- Don't split on special char start/ending " ' ` - _
    - Don't split when it starts with "St"

    Added tests cases & updated annotations

    Updated batch detected entities after fixing name splitting

### CODE REVIEW

File: tests/test_detect_entities.py [hunk 1]
```diff
@@ -35,3 +35,67 @@ def test_split_person():
     assert entities[1]["type"] == "PERSON"
     assert not entities[1]["in_source"]
     assert summary[entities[1]["start"] : entities[1]["end"]] == entities[1]["ent"]
+
+
+def test_split_person_apostrophe():
+    summary = "A search is under way for Daniel Levenson's cat"
```
Comment (dleve123) [inline tests/test_detect_entities.py:41]: 🐱

File: tests/test_detect_entities.py [hunk 2]
```diff
@@ -35,3 +35,67 @@ def test_split_person():
...
+    assert not entities[0]["in_source"]
+    assert summary[entities[0]["start"] : entities[0]["end"]] == entities[0]["ent"]
+
+    assert entities[1]["ent"] == '"Brad"'
+    assert entities[1]["type"] == "PERSON"
+    assert not entities[1]["in_source"]
+    assert summary[entities[1]["start"] : entities[1]["end"]] == entities[1]["ent"]
+
+
+def test_split_person_dash():
```
Comment (dleve123) [inline tests/test_detect_entities.py:78]: This is the test that would have failed without this behavior change, correct? The others tests are just backfilling?

File: src/detect_entities.py [hunk 3]
```diff
@@ -1,33 +1,41 @@
...
+        split_entity.append(
+            {
+                "ent": sub_ent,
+                "type": "PERSON",
+                "start": start_idx,
+                "end": start_idx + len(sub_ent),
+                "in_source": is_entity_contained(sub_ent, source),
+            }
+        )
+        start_idx += len(sub_ent) + 1
```
Comment (dleve123) [inline src/detect_entities.py:23]: I guess the `+ 1` handles the fact that the multiple parts of a name can't overlap / are separated by whitespace? In the removed diff, it seems like we're always using spacy's `entity.idx` to get the starting index.

Why not continue to leverage spacy here? I feel like the less manual handling of indices, the better.

UPDATE: Ah, I think this is because we're taking a single multi-word span, i.e. `Daniel Levenson` and now handling it directly as a `List[str]`, is that right?

File: src/detect_entities.py [hunk 4]
```diff
@@ -1,33 +1,41 @@
...
+from spacy.tokens import Span
 from src.entity_utils import MarkedEntity, is_entity_contained
 
-
 nlp = spacy.load("en_core_web_lg")
 
 
+def split_person_entity(entity: Span, source: str) -> List[MarkedEntity]:
+    split_entity: List[MarkedEntity] = []
+    ent_parts = entity.text.split(" ")
```
Comment (dleve123) [inline src/detect_entities.py:11]: minor:

```suggestion
    ent_parts: List[str] = entity.text.split(" ")
```

just think it's makes the rest of the function easier to scan.

File: results/test-extrinsic-100-summaries.json [hunk 5]
```diff
@@ -1 +1 @@
...:0,"28071499":0,"28772094":0,"29252950":0,"30457745":0,"30657223":0,"30760966":0,"31113635":0,"32033856":0,"32132840":0,"32487884":0,"32557344":0,"32651759":0,"32798693":0,"33044932":0,"33106194":0,"33317600":0,"33648595":0,"34193568":0,"34427234":0,"34474416":0,"34577604":0,"34614991":0,"34660292":0,"34694547":0,"34760893":0,"34787804":0,"34898778":0,"35038754":0,"35159263":0,"35237411":0,"35305028":0,"35308608":0,"35360415":0,"35369658":0,"35411078":0,"35698210":0,"35723757":0,"35811486":0,"35845462":0,"35972298":0,"36000903":0,"36061830":0,"36066252":0,"36162485":0,"36295288":0,"36301742":0,"36327891":0,"36341722":0,"36540012":0,"36555657":0,"36635880":0,"36730443":0,"36772094":0,"36874565":0,"36912060":0,"36932519":0,"36961338":0,"37310327":0,"37327555":0,"37391662":0,"37603756":0,"37604251":0,"37870098":0,"37961975":0,"38003317":0,"38078754":0,"38113695":0,"38219213":0,"38299022":0,"38399324":0,"38414086":0,"38470818":0,"38779380":0,"38983532":0,"39431070":0,"39535308":0,"39556625":0,"39582812":0,"39593375":0,"39753996":0,"39784504":0,"39807248":0,"39847765":0,"39895178":0,"39959265":0,"40243246":0,"40285040":0,"40306866":0,"40426136":0,"40498102":0,"40551256":0,"40587903":0}}
```
Comment (dleve123) [inline results/test-extrinsic-100-summaries.json:1]: I feel like JSONL would lead to better diffs! Hard to parse the change, but /shrug

**Review by dleve123 — COMMENTED** [review]
@anton164 Can you briefly comment on the commands ran to update these files? Seems like this change only impacted BART-Large baseline and the corrector models' performance? Kind of surprising that it didn't impact FBS at all?

File: src/detect_entities.py (hunk 3 above)
Comment (anton164) [reply src/detect_entities.py:23]: Your last comment is correct - we now split by space instead of using spacy to split into sub entites.

File: tests/test_detect_entities.py (hunk 2 above)
Comment (anton164) [reply tests/test_detect_entities.py:78]: _dash, _apostrophe, _quote and _st were all unsupported prior to this PR, altho I had added manual handling for some of them without tests.

**Comment by anton164** [comment]
@dleve123 this change didn't impact performance of any model. It just changed the `ents_per_sum` slightly for corrector and bart large baseline. These are edge cases - none of the FBS summaries contained these kinds of entities.

**Comment by dleve123** [comment]
> @dleve123 this change didn't impact performance of any model. It just changed the ents_per_sum slightly for corrector and bart large baseline. These are edge cases - none of the FBS summaries contained these kinds of entities.

Gotcha, makes sense! Approving PR – reviewing the larger one now.

**Review by dleve123 — APPROVED** [review]

File: tests/test_detect_entities.py (hunk 2 above)
Comment (anton164) [reply tests/test_detect_entities.py:78]: Correct!
